### PR TITLE
MAINT fft

### LINKED ISSUE #386: Inverse fft in 2D backend
EDIT by EO:
- [x] Using the appropriate normalization for the ifft (ie both backend files)
- [x] Adding a unit test in 2d/test for testing the fft, which is not currently tested 
- [x] Normalizing the wavelet in the Fourier domain (ie the file filter_bank.py) to avoid the new code to break and satisfy the previous unit tests
NOTE: this might lead to a small(not to say tiny) speed-up, because the dot-wise multiplication usually use quite some resources!

In 2D backend code, in
`            output = torch.ifft(input, 2, normalized=False)*input.size(-2)*input.size(-3)
`
why is the inverse fft routine multiplied by the term *input.size(-2)*input.size(-3)? I was expecting that transforming a matrix back and forth to the Fourier domain give the same result. In other words, I was expecting outputs of the following test code match.

```
import torch
from kymatio.scattering2d import backend as bd2

pad = bd2.Pad(2 ** 0, pre_pad=False)

# -- parameters
M = 8
N = 8
L = 8
J = 4

# -- synthesise data
y = torch.rand(1, 1, M, N)
for i in range(M):
    for j in range(N):
        y[:,:,i,j] = i+j

y_pad = pad(y)                                    # pad to add complex dimension
y_fft = bd2.fft(y_pad, 'C2C')                     # fft
y_fft_inv = bd2.fft(y_fft, 'C2C', inverse=True)   # inverse fft
y_fft_inv_un = y_fft_inv[..., 1:-1, 1:-1, 0]      # unpad

# -- print
print(y)
print(y_fft_inv_un)
```

### PR DESCRIPTION
**Pull request by edouardoyallon**
Close #386

### CODE REVIEW

**Comment by edouardoyallon** [comment]
> Before we normalized, now we do not. How come?

So - this will be a more standard fft, even if it doesn't satisfy fft o ifft = Id. I think it's more standard this way, than before.

@janden view?

File: kymatio/scattering2d/tests/test_numpy_scattering2d.py [hunk 1]
```diff
@@ -149,10 +149,14 @@ def test_fft(self, backend):
 
         z = backend.fft(x, direction='C2C', inverse=True)
 
+        z = z * 4
```
Comment (lostanlen) [inline kymatio/scattering2d/tests/test_numpy_scattering2d.py]: i would prefer if you didn't update the z between the definition and the test

**Comment by lostanlen** [comment]
`ifft(fft(x))` is an absolute must

**Comment by edouardoyallon** [comment]
> > So - this will be a more standard fft, even if it doesn't satisfy fft o ifft = Id. I think it's more standard this way, than before.
> 
> It does satisfy `ifft(fft(x)) = x`, as far as I can tell. Why wouldn't it? It's just not unitary.

?? No, I had to renormalize by "4", cf test

File: kymatio/scattering2d/tests/test_numpy_scattering2d.py (hunk 1 above)
Comment (edouardoyallon) [reply kymatio/scattering2d/tests/test_numpy_scattering2d.py]: @janden see

**Comment by janden** [comment]
> ?? No, I had to renormalize by "4", cf test

When you test the `ifft`, yes. In that test, `y` is the DFT of `x`. Since this is a 2×2 DFT, it also happens to be `4*ifft(x)`, so we simplify the test by comparing to that. Might be less confusing to simply take `ifft(y)` and compare it to `x`.

**Comment by edouardoyallon** [comment]
> > ?? No, I had to renormalize by "4", cf test
> 
> When you test the `ifft`, yes. In that test, `y` is the DFT of `x`. Since this is a 2×2 DFT, it also happens to be `4*ifft(x)`, so we simplify the test by comparing to that. Might be less confusing to simply take `ifft(y)` and compare it to `x`.

damn, I read too quickly - then in this case, there is an inconsistency in the normalization of the filters.. I'm ok to leave as it is.

**Comment by janden** [comment]
> I'm ok to leave as it is.

You mean close the PR? Why? Can't we just properly normalize the filters?

**Comment by edouardoyallon** [comment]
> > I'm ok to leave as it is.
> 
> You mean close the PR? Why? Can't we just properly normalize the filters?

Sorry, it's late here, I'm alwaysa  bit confused by the normalization linked to the downsmapling, it's actually all good, sorry for the mess
